openshift-kube-apiserver/filters/deprecatedapirequest/crd_check.go

```diff
@@ -0,0 +1,59 @@
+package deprecatedapirequest
+
+import (
+	"context"
+	"net/http"
+	"sync"
+	"time"
+
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apiextensionsinformers "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+)
+
+const CrdEstablished ContextKey = "DeprecatedApiRequestCrdEstablished"
+
+// WithCrdCheck adds a filter to the hander chain that set the CrdEstablished context key to true if the
+// deprecatedapirequests.apiserver.openshift.io CRD has been installed.
+func WithCrdCheck(handler http.Handler, config *rest.Config) http.Handler {
```

Comment:
reading top to bottom, it seems weird that we need to do this.  I would expect a general flow of

1. filter watches for access
2. on an interesting access (in the future all accesses), make a mark for a controller
3. at some asynchronous time, the controller attempts to write.
4. if the write fails, the controller retries eventually.

in such a flow, checking for the presence of a CRD doesn't matter.